File: sample-new-resource-detector/src/test/java/nl/haploid/octowight/sample/TestData.java
package nl.haploid.octowight.sample;

import nl.haploid.octowight.AtomChangeEvent;
import nl.haploid.octowight.registry.data.ResourceRoot;
import nl.haploid.octowight.sample.repository.PersonDmo;
import nl.haploid.octowight.sample.repository.RoleDmo;

import java.util.Random;
import java.util.UUID;

public class TestData {

	public static AtomChangeEvent atomChangeEvent() {
		return atomChangeEvent(nextLong());
	}

	public static AtomChangeEvent atomChangeEvent(final long atomId) {
		final AtomChangeEvent event = new AtomChangeEvent();
		event.setAtomId(atomId);
		event.setAtomLocus(name());
		event.setAtomType(PersonDmo.ATOM_TYPE);
		return event;
	}

	public static String name() {
		return UUID.randomUUID().toString();
	}

	public static long nextLong() {
		return new Random().nextLong();
	}

	public static PersonDmo personDmo() {
		return personDmo(nextLong());
	}

	public static PersonDmo personDmo(final long id) {
		final PersonDmo dmo = new PersonDmo();
		dmo.setId(id);
		dmo.setName(name());
		return dmo;
	}

	public static ResourceRoot resourceRoot(final Long resourceId) {
		final ResourceRoot resourceRoot = new ResourceRoot();
		resourceRoot.setResourceId(resourceId);
		resourceRoot.setResourceType("olson");
		resourceRoot.setAtomId(nextLong());
		resourceRoot.setAtomLocus("madison avenue");
		resourceRoot.setAtomType("draper");
		return resourceRoot;
	}

	public static RoleDmo roleDmo(final PersonDmo personDmo, final String type) {
		final RoleDmo dmo = new RoleDmo();
		dmo.setId(nextLong());
		dmo.setPerson(personDmo);
		dmo.setType(type);
		return dmo;
	}

	public static String topic() {
		return UUID.randomUUID().toString();
	}
}

File: sample-new-resource-detector/src/test/scala/nl/haploid/octowight/sample/TestData.scala
package nl.haploid.octowight.sample

import java.util.{Random, UUID}

import nl.haploid.octowight.AtomChangeEvent
import nl.haploid.octowight.registry.data.{Atom, ResourceRoot}
import nl.haploid.octowight.sample.repository.{PersonDmo, RoleDmo}

object TestData {

  def atomChangeEvent: AtomChangeEvent = atomChangeEvent(nextLong)

  def atomChangeEvent(atomId: Long) =
    new AtomChangeEvent(id = nextLong, atomId = atomId, atomOrigin = nextString, atomCategory = PersonDmo.AtomCategory)

  def name = nextString

  def nextLong: Long = new Random().nextLong

  def personDmo: PersonDmo = personDmo(nextLong)

  def personDmo(id: Long) = {
    val dmo = new PersonDmo
    dmo.setId(id)
    dmo.setName(name)
    dmo
  }

  def resourceRoot(resourceId: Long) =
    new ResourceRoot(resourceId = resourceId, resourceType = nextString, root = atom, version = null)

  def atom = new Atom(nextLong, nextString, nextString)

  def roleDmo(personDmo: PersonDmo, name: String) = {
    val dmo = new RoleDmo
    dmo.setId(nextLong)
    dmo.setPerson(personDmo)
    dmo.setName(name)
    dmo
  }

  def topic = nextString

  def roleDmo(id: Long): RoleDmo = {
    val dmo = new RoleDmo
    dmo.setId(id)
    dmo
  }

  def nextString = UUID.randomUUID.toString
}
